tests/unit/graph_reader/test_artifact_collection.py

```diff
@@ -174,3 +176,55 @@ def test_two_sessions(
                 assert art_pred_session_line_index >= 0
                 # required art session is in from of art session
                 assert art_pred_session_line_index < art_session_line_index
+
+
+@pytest.mark.parametrize(
+    "input_parameters, input_values, expected_values",
+    [
+        pytest.param([], dict(), {"add_p": "ap"}, id="no_input"),
+        pytest.param(["a", "p"], dict(), {"add_p": "ap"}, id="default_value"),
+        pytest.param(
+            ["a", "p"],
+            {"a": "A", "p": "P"},
+            {"add_p": "AP"},
+            id="overriding_value",
+        ),
+    ],
+)
+def test_module_run(
+    linea_db, execute, input_parameters, input_values, expected_values
+):
+    """
+    Test the module generated from ArtifactCollection can run with/without
+    input parameters
+    """
+
+    code = """\n
+import lineapy
+a = "a"
+p = "p"
+b = a+p
+lineapy.save(b,'add_p')
+"""
+    execute(code, snapshot=False)
+    artifact_list = ["add_p"]
+    ac = ArtifactCollection(
+        linea_db, artifact_list, input_parameters=input_parameters
+    )
+    temp_folder = tempfile.mkdtemp()
+    temp_module_path = pathlib.Path(
+        temp_folder, "artifactcollection_module.py"
+    )
```

Comment:
should wrap this in a context managed block for cleanup. alternately,
pytest also gives you a fixture called tmp_path. that way you won't have to worry about any cleanup either. we already use this in the project if you search for the exact text of tmp_path you should find a few tests.